index.ts

```diff
@@ -1,35 +1,60 @@
 export class OatyArray {
-  private _original: object[]
-  private _keys: string[]
-  private _transposed: object = {}
+  private _keys: string[] | undefined
+  private _original: object[] = []
+  private _transposed: { [key: string]: any } = {}
+
+  constructor(data?: object[], options?: { keys?: string[] }) {
+    if (data !== undefined) { this.push(...data) }

```

Comment:
these `undefined` checks are still unnecessary and should be rewritten imo, for example, this line can just be:

```ts
this.push(...data || [])
```

Reply:
Then I may as well use defaults and do
`constructor(data: object[] = [], options?: { keys?: string[] }) {
    this.push(...data)`

Reply:
Imo `if (data !== undefined)` is more readable than having a default

Reply:
I think default params to keep it concise

Reply:
Okay

Reply:
Outdated. Data is an optional parameter again